app/application/use_cases/add_vault_member.py

```
@@ -132,6 +135,10 @@ def execute(self, inp: AddVaultMemberInput) -> AddVaultMemberResult:
                 inp.role
             )
             if updated is None:
+                
+                # Track member role update
+                track_vault_member_added(role=inp.role.value)
+                
                 raise RuntimeError("Authorization missing during role update")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)